File: leetcode/java/1773. Count Items Matching a Rule.java
class Solution {
    public int countMatches(List<List<String>> items, String ruleKey, String ruleValue) {
        int findIndex = 0;
        
        switch(ruleKey) {
            case "type":
                findIndex = 0;
                break;
            case "color":
                findIndex = 1;
                break;
            default:
                findIndex = 2;
        }

        int count = 0;

        for(List<String> element : items) {
            if(element.get(findIndex).equals(ruleValue)) {
                count++;
            }
        }

        return count;
    }
}

File: leetcode/kotlin/1773. Count Items Matching a Rule.kt
class Solution {
    fun countMatches(items: List<List<String>>, ruleKey: String, ruleValue: String): Int {
        val findIndex = when(ruleKey) {
            "type" -> 0
            "color" -> 1
            else -> 2
        }
        
        var count = 0
        items.forEach {
            count += if(it.get(findIndex) == ruleValue) 1 else 0 
        }
        
        return count
    }
}
